src/nimblepkg/download.nim

```diff
@@ -309,6 +309,26 @@ proc doDownloadTarball(url, downloadDir, version: string, queryRevision: bool):
     raise nimbleError(tryDoCmdExErrorMessage(cmd, output, exitCode))
   display("Completed", "unpacking " & filePath)
 
+  when defined(windows):
+    # On Windows symbolic link files are not being extracted properly by the
+    # `tar` command. They are extracted as empty files, but when cloning the
+    # repository with Git they are extracted as ordinary files with the link
+    # path in them. For that reason here we parse the tar file content to
+    # extract the symbolic links and add their paths manually to the content of
+    # their files.
+    let listCmd = &"{getTarExePath()} -ztvf {filePath} --force-local"
+    let (cmdOutput, cmdExitCode) = doCmdEx(listCmd)
+    if cmdExitCode != QuitSuccess:
+      raise nimbleError(tryDoCmdExErrorMessage(listCmd, cmdOutput, cmdExitCode))
+    let lines = cmdOutput.splitLines()
+    for line in lines:
+      if line.contains(" -> "):
+        let parts = line.split
+        let linkPath = parts[^1]
+        let linkNameParts = parts[^3].split('/')
+        let linkName = linkNameParts[1 .. ^1].foldl(a / b)
+        writeFile(downloadDir / linkName, linkPath)
+
```

Comment:
This sounds like we will always have edge cases where our hashes are different. Maybe we should instead just ignore symlinks in our hash calculations? How do other package managers handle this?

Reply:
I think that for now all edge cases are handled properly.

Reply:
Also, filtering the symbolic links will be harder to implement because where the package's checksum is calculated I already don't have the tar file. The information has to be moved through a few call stack levels in some data structure which will be used only by a part of the code paths. It will lead to a much uglier code IMHO.